src/services/network/mod.rs

```diff
@@ -273,31 +427,63 @@ impl NetworkService {
                     State::Error
                 }
             },
-            State::Active(conn) => {
+            State::Active(conn, choice) => {
                 info!("Listening for network events");
 
-                match NetworkService::events(&conn).await {
-                    Ok(mut events) => {
-                        while let Some(event) = events.next().await {
-                            let mut exit_loop = false;
-                            if let NetworkEvent::WirelessDevice { .. } = event {
-                                exit_loop = true;
+                // TODO: i dont know how to combine the opaque types.. rust streams
+                match choice {
+                    BackendChoice::NetworkManager => {
+                        let nm = NetworkDbus::new(&conn).await.unwrap();
+
+                        match nm.subscribe_events().await {
```

Comment:
```suggestion
                        match NetworkDbus::new(&conn).await {
                          Ok(nm) => match nm.subscribe_events().await {
                          ...
                                                      Err(err) => {
                                error!("Failed to connect to NetworkManager: {}", err);
                                State::Error
                            }
```

Same thing for the IWD case